### PR TITLE
Access control for rpc request over tcp

### PR DESCRIPTION
**Pull request by hairtail**
## Summary
Add auth system for protected rpc requests over tcp. With this PR, protected routes can be visited with a specific token.
## Testing Plan
Local test completed.
## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/fileStores/config.ts [hunk 1]
```diff
@@ -330,6 +336,7 @@ export class Config extends KeyStore<ConfigOptions> {
       jsonLogs: false,
       explorerBlocksUrl: DEFAULT_EXPLORER_BLOCKS_URL,
       explorerTransactionsUrl: DEFAULT_EXPLORER_TRANSACTIONS_URL,
+      authToken: DEFAULT_TCP_AUTH_TOKEN,
```
Comment (NullSoldier) [inline ironfish/src/fileStores/config.ts:339]: We should name this `rpcAuthToken`

Comment (NullSoldier) [reply ironfish/src/fileStores/config.ts:339]: The name of this default doesn't match the config.

Comment (hairtail) [reply ironfish/src/fileStores/config.ts]: Done.

**Comment by NullSoldier** [comment]
Were going to credit you for this PR but were gonna close this in favor of https://github.com/iron-fish/ironfish/pull/2224 which we were already starting. Thank you for this PR because it's serving as the base for their PR.

**Comment by hairtail** [comment]
> Were going to give you testnet credit for this PR, but were gonna close this in favor of #2224. We already started it internally, and want to do it slightly differently. Mostly we want to
> 
> 1. Have authentication live in RpcServer and allow any adapter to use it
> 2. Authenticate over the whole adapter and delete Protected Namespaces (similar to bitcoin)

Sorry, i can't get the idea why we need auth over the whole adapter. There are some public rpc interface like `chain/getBlock, node/getStatus, fee/getFees`. These kind of routes are unnecessary to verify the auth.

**Pull request closed without merging**